src/java.base/share/classes/java/security/DigestInputStream.java

```diff
@@ -51,6 +51,15 @@
  * retain a handle onto the digest object, and clone it for each
  * digest to be computed, leaving the original digest untouched.
  *
+ * @implNote This implementation updates and only updates the message digest
+ *      with data actually read from the input stream when it is turned on.
```

Comment:
"when it is turned on"?  You could probably just end with "actually read from the input stream."